doc/rst/tools/chpl-language-server/chpl-language-server.rst

```diff
@@ -0,0 +1,193 @@
+       single_file_support = true,
+       root_dir = util.find_git_ancestor,
+       settings = {},
+     },
+   }
+   
+   lspconfig.cls.setup{}
+   vim.cmd("autocmd BufRead,BufNewFile *.chpl set filetype=chpl")
+
+
+VSCode
+^^^^^^
+
+Install the soon-to-be-released ``chapel`` extension from the `Visual Studio
+Code marketplace
+<https://marketplace.visualstudio.com/items?itemName=chpl-hpe.chapel>`_.
+
+.. note::
+
+   The extension is not yet available at the time of writing and the above link
+   may not work until then. This section will be updated when it is available.
+
+Supported Features
+------------------
+
+``CLS`` supports a number of core features, with further features controlled by configuration options.
+
+.. note::
+
+   Not all editor integrations may support all features. This list details
+   features that are supported by ``CLS`` itself, not by any particular editor.
+
+Core Features
+^^^^^^^^^^^^^
+
+* Diagnostics
+   Errors and warnings are reported as you type. ``CLS`` also has limited
+   support to fix some errors and warnings automatically with quick fixes.
+* Symbol Information
+   ``CLS`` provides basic information about symbols in code, enabling various
+   actions.For example, you can jump to the definition of a symbol, find all
+   references to a symbol, and rename a symbol. Hovering over a symbol will
+   show its definition and documentation. Selecting a symbol will highlight all
+   references to it in the file.
+* Code Completion
+   ``CLS`` provides limited code completion for symbols in your code.
+
+Optional Features
+^^^^^^^^^^^^^^^^^
+
+* End of Block Markers
+   ``CLS`` can display markers after a closing brace to indicate what block it
+   closes. This feature is off-by-default, but can be enabled with
+   ``--end-markers <marker_list>``.
+
+   ``<marker_list>`` is a comma-separated list of markers to display. The
+   following markers are supported:
+
+      * ``none``: Do not display any end markers. This is the default.
+      * ``all``: Display all end markers.
+      * ``decl``: Display markers for the end of a declaration (e.g. modules,
+        records, functions, etc.)
+      * ``loop``: Display markers for the end of a loop (e.g. ``for``,
+        ``forall``, ``while``, etc.)
+      * ``block``: Display markers for the end of other important blocks (e.g.
+        ``on``, ``begin``, etc.)
+
+   End of block markers have a threshold of 10 lines of code. If a block is
+   smaller than this, no end marker will be displayed. This threshold can be
+   adjusted with ``--end-marker-threshold <threshold>``.
+
+Experimental Resolver Features
+^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
+
+All of the following features are disabled without the ``--resolver`` flag. To
+use them, enable ``--resolver``.
+
+.. warning::
+
+   These features rely on the Dyno resolver, which is not finished and actively
+   under development. It is not unexpected for ``--resolver`` to cause ``CLS``
+   to crash or hang.
```

Comment:
double negative on 'not unexpected'. I would say "it's fairly common for `--resolver` to cause `CLS` to crash or hang".